src/trace/layers/ordered.rs

```diff
@@ -38,36 +39,37 @@ impl<K: Ord+Clone, L: Trie> Trie for OrderedLayer<K, L> {
 			let child_lower = self.offs[lower];
 			let child_upper = self.offs[lower + 1];
 			OrderedCursor {
-				keys: self.keys.clone(),
-				offs: self.offs.clone(),
+				keys: owned_self.clone().map(|x| &x.keys),
+				offs: owned_self.clone().map(|x| &x.offs),
```

Comment:
@frankmcsherry Regarding the use of `Deref<Target=Self>` the problem seems to be that I wouldn't know how to express this: our constraint is that the "thing that dereferences to Self" also supports grabbing another reference that's reachable from `Self` (in this case what's expressed by the closure).